better docs (#16)
TypeScriptToLua/TypeScriptToLua.github.io: docs/advanced/writing-declarations.md

### Operator Overloads

> Lua supports overloading of mathematical operators such as `+`, `-` or `*`. (This is done via adding a function to `__add`, `__sub`, `__mul`, and so forth.) Since TypeScript does not support operator overloading in its type system, this feature is hard to replicate. Unfortunately, this is not something that can be fixed properly right now without forking off our custom TypeScript version.

Reviewer: > adding a function to `__add`, ...

That sounds like you'd add these functions to the table itself, not it's metatable. Maybe something like

```suggestion
Lua supports overloading of mathematical operators such as `+`, `-` or `*`, using [metatable methods](https://www.lua.org/manual/5.4/manual.html#2.4) such as `__add`, `__sub` or `__mul`. Since TypeScript does not support operator overloading in its type system, this feature is hard to replicate. Unfortunately, this is not something that can be fixed properly right now without forking off our custom TypeScript version.
```

Not sure what would be the best link here
- https://www.lua.org/manual/5.4/manual.html#2.4
- https://www.lua.org/pil/13.1.html
- http://lua-users.org/wiki/MetamethodsTutorial

Author: link to <https://www.lua.org/pil/13.1.html> seems fine.
otherwise, LGTM!

Reviewer 2: I am not sure what the point is of adding this detail, If we don't support overloading anyway. Seems a bit off topic to me

Author: I think the point is that so that when someone searches the docs for "__add" or "__mul", the relevant section pops up.

Author: Otherwise there are 0 hits.
